### PR TITLE
Add hasura GQL API

### PR DESCRIPTION
**Pull request by clostao**
This PR implements Hasura GQL API:

- Adds to the needed metadata for tracking the all the tables needed in Auto-Drive Frontend
- Adds two roles `user` & `app-admin`. The first is the base permission and the second it's assigned to users with the role `Admin`.
- Implements Hasura Claims into Custom JWT auth provider
- Updates how the app is deployed in both production and local

### CODE REVIEW

**Review by marc-aurele-besner — COMMENTED** [review]
This is awesome work 🙌 
Just one comment to add the auth directly in ApolloClient

File: frontend/src/services/gql/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { ApolloClient, InMemoryCache } from '@apollo/client';
+
+export const apiv2Client = new ApolloClient({
+  uri: process.env.NEXT_PUBLIC_GQL_URL,
+  cache: new InMemoryCache(),
```
[the remark this answers is not in the document]
Comment (clostao) [reply frontend/src/services/gql/index.ts:5]: The session token refreshed quite frequently (most times the user interacts with nextjs server). Do you mean that there's a dynamic to do so?

[the remark this answers is not in the document]
Comment (marc-aurele-besner) [reply frontend/src/services/gql/index.ts:5]: We could also look into implementing https://www.npmjs.com/package/@apollo/experimental-nextjs-app-support so the ApolloClient to work on both SSR and RSR.

[the remark this answers is not in the document]
Comment (clostao) [reply frontend/src/services/gql/index.ts:22]: It worked thanks for the comment & review!

[the remark this answers is not in the document]
Comment (clostao) [reply frontend/src/services/gql/index.ts:22]: Update: It worked fine on `yarn dev` mode but compiling the nextjs app it failed due to [this error](https://nextjs.org/docs/messages/dynamic-server-error). 

TLDR; Static generation of routes is checked through the compiler discarding the ones depending on dynamic values (such as `cookies()` or `headers()`). The issue is that the compiler is unable to detect if a route is dynamic or static when the call to these dynamic values is at another [AsyncContext](https://nextjs.org/docs/messages/dynamic-server-error#what-is-async-context) than the page. In this case fails in build-time.

For solving this, you have to explicitly determine that the route is dynamic

**Review by marc-aurele-besner — APPROVED** [review]
AWESOME :)

**Review by marc-aurele-besner — APPROVED** [review]
